### PR TITLE
Replace Magick with Rubick for RMRK

### PR DESCRIPTION
**Pull request by vikiival**
### What is this

This PR replaces magick (SubQuery) with rubick (SubSquid).
It is a step towards more reusable / chain agnostic components.

### Did I break something

Hope not that's why we have @kodadot/qa-guild

### Is there some work that needs to be done? 

Yes I skipped some components. 

Screenshot 2022-08-01 at 15 10 26

### Commit changelog

- :alien: collectionFor mint for RMRK
- :zap: CreateToken is using subquery
- :bug: :label: scroll mixin should return boolean | undefined
- :zap: using this.client instead of ternary op
- :coffin: removed dead getter
- :zap: Gallery is using :squid:
- :coffin: passion feed can come later again
- :bug: profile activity shows NaN if no NFT was sold
- :broom: optimize imports
- :alien: query to get only symbols for user
- :zap: simplemint is using :squid:
- :bug: WARNING: heuristic fragment matching going on!
- :alien: :squid: nftById for RMRK
- :alien: query all events by nftId
- :zap: collection on NFT is object
- :zap: GalleryItem on Squid
- :zap: fetch if NFT is not burned in the meantime
- :zap: check before buy is using :squid:

**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [x] Bugfix
- [x] Feature
- [x] Refactoring

### What's new?

- [ ] PR closes #<issue_number>
- [ ] Requires deployment <rubick/magick_issue>
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [ ] My contribution builds **clean without any errors or warnings**
- [ ] I've merged recent default branch -- **main** and I've no conflicts
- [ ] I've tried to respect high code quality standards
- [ ] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [ ] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=<My_Kusama_Address_check_https://github.com/kodadot/nft-gallery/blob/main/CONTRIBUTING.md#creating-your-ksm-address>)

### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

File: components/rmrk/Gallery/Gallery.vue [hunk 1]
```diff
@@ -169,7 +160,7 @@ export default class Gallery extends mixins(
   }
 
   get isRmrk(): boolean {
-    return this.urlPrefix === 'rmrk' || this.urlPrefix === 'westend'
+    return false
   }
```
Comment (Jarsen136) [inline components/rmrk/Gallery/Gallery.vue:164]: I would like to remove `isRmrk` function and also `this.isRmrk` in this file

**Review by Jarsen136 — COMMENTED** [review]
small stuff

File: components/rmrk/Gallery/Gallery.vue (hunk 1 above)
Comment (vikiival) [reply components/rmrk/Gallery/Gallery.vue]: Thanks for the comment! should be resolved

**Review by Jarsen136 — APPROVED** [review]
code LGTM

**Comment by petersopko** [comment]
- [x] minted NFT
- [x] minted collection
- [x] browsed gallery with no problem
- [x] browsed collections with no problem
- [x] detail of gallery item works
- [x] detail of collection works
- [x] new NFT loads in gallery after mint
- [x] burning NFT works
- [x] new collection loads in dropdown select a little while after mint ( explainer incoming https://github.com/kodadot/nft-gallery/pull/3627)

Previous problems are gone, this works for me.

**Review by petersopko — APPROVED** [review]
wfm

**Comment by JustLuuuu** [comment]
STATS: Offers numbers are a little bit off I think. 
image

**Pull request merged**